src/config.js

```diff
@@ -49,6 +48,40 @@ export async function getConfig() {
   return _config
 }
 
+/**
+ * @param {string} file
+ * @returns {Promise<LazyConfig>}
+ */
+async function loadConfigFromFile(file) {
+  if (file.endsWith('.json')) {
+    return JSON.parse(readFileSync(file, 'utf8'))
+  } else if (file.endsWith('.ts') || file.endsWith('.mts') || file.endsWith('.cts')) {
+    const { build } = await import('esbuild')
+    const result = await build({
+      absWorkingDir: workspaceRoot,
+      entryPoints: [file],
+      outfile: 'out.js',
+      target: 'esnext',
+      platform: 'node',
+      format: 'esm',
+      write: false,
+    })
+    const { text } = result.outputFiles[0]
```

Comment:
I guess it's not necessary to handle empty array here because it's not possible to get it.